compute_endpoint/globus_compute_endpoint/endpoint/config/config.py

```diff
@@ -253,15 +268,25 @@ def __init__(
         self.stderr = stderr
 
     @property
-    def executors(self) -> tuple[GlobusComputeEngineBase, ...]:
-        return self._executors
+    def engine(self) -> GlobusComputeEngineBase | None:
+        return self._engine
+
+    @engine.setter
+    def engine(self, val: GlobusComputeEngineBase | None):
+        self._engine = val
+
+    @property
+    def executors(self) -> tuple[GlobusComputeEngineBase | None]:
+        warnings.warn("`executors` is deprecated; use `engine`", stacklevel=2)
+        return (self.engine,)
 
     @executors.setter
     def executors(self, val: t.Iterable[GlobusComputeEngineBase] | None):
+        warnings.warn("`executors` is deprecated; use `engine`", stacklevel=2)
         if val is None:
-            self._executors: tuple[GlobusComputeEngineBase, ...] = ()
+            self.engine = GlobusComputeEngine()
         else:
-            self._executors = tuple(e for e in val)[:1]  # enforce len() == 0 or 1
+            self.engine = tuple(e for e in val)[0]
```

Comment:
since we're setting `engine` here, suggest moving all validation/slicing of executor lists to this method

Reply:
I actually wrote this in early December, so that particular detail has paged out of my brain.  I vaguely recall there's a reason for doing it this way, perhaps to-do with mypy?  I'll page back in that detail and follow-up in this thread.

Reply:
Yep; it's a mypy-thing.  I tried to do exactly as you've suggested, and this was as close as I could get.  Too bad this isn't Rust, with a much smarter type-system and compiler.  "Just sayin'"